.github/workflows/autoupdate-license.yaml

```diff
@@ -0,0 +1,27 @@
+name: Update License Year
+
+on:
+  schedule:
+    - cron: '0 0 1 1 *'
+  push:
+    branches: ["main"]
+
+jobs:
+  update-year:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Update year in license file
+        run: |
+          year=$(date +%Y)
+          sed -i "s/2017-[0-9]*/2017-$year/g" LICENSE
+
+      - name: Set git config
+        run: |
+          git config user.name "Automated"
+          git config user.email "actions@users.noreply.github.com"
+          git add LICENSE
+          git commit -m "Update license years" || exit 0
+          git push
```

Comment:
Does this action really have permissions to commit any changes to the repository?

Reply:
In case you allow "Read and write permissions" - yes. [Docs](https://docs.github.com/en/enterprise-server@3.6/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#configuring-the-default-github_token-permissions)